libtiledbsoma/src/soma/managed_query.cc

```diff
@@ -147,11 +147,21 @@ void ManagedQuery::setup_read() {
 
     // Allocate and attach buffers
     LOG_TRACE("[ManagedQuery] allocate new buffers");
-    buffers_ = std::make_shared<ArrayBuffers>();
-    for (auto& name : columns_) {
-        LOG_DEBUG(fmt::format("[ManagedQuery] [{}] Adding buffer for column '{}'", name_, name));
-        buffers_->emplace(name, ColumnBuffer::create(array_, name));
-        buffers_->at(name)->attach(*query_);
+    if (!buffers_) {
```

Comment:
I haven't fully dug into how the buffers are being shared in the Python/R layers. Will reusing the same buffers be a problem for multi-chunk reads?